Soccar [Physics]/src/nl/soccar/physics/models/CarPhysics.java

```diff
@@ -125,6 +131,33 @@ private void updateSteerAngle() {
         }
     }
 
+    private void updateBoost() {
+
+        // Only allow boosting when the trail is gone.
+        if (car.getThrottleAction() == ThrottleAction.BOOST && trail.size() == 0) {

```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Use isEmpty() to check whether the collection is empty or not. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AS1155)